File: app/src/main/java/in/sudopk/vaishnavacalendar/ButtonDescription.java
package in.sudopk.vaishnavacalendar;

import android.os.Parcel;
import android.os.Parcelable;

public class ButtonDescription implements Parcelable {
    public final String title;
    public final String id;

    public ButtonDescription(String title, String id) {
        this.title = title;
        this.id = id;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(final Parcel dest, final int flags) {
        dest.writeString(title);
        dest.writeString(id);
    }

    public static final Parcelable.Creator<ButtonDescription> CREATOR
            = new Parcelable.Creator<ButtonDescription>() {
        public ButtonDescription createFromParcel(Parcel in) {
            return new ButtonDescription(in);
        }

        public ButtonDescription[] newArray(int size) {
            return new ButtonDescription[size];
        }
    };

    private ButtonDescription(Parcel in) {
        title = in.readString();
        id = in.readString();
    }
}

File: app/src/main/java/in/sudopk/vaishnavacalendar/ButtonDescription.kt
package `in`.sudopk.vaishnavacalendar

import android.os.Parcel
import android.os.Parcelable

class ButtonDescription(val id: String, val title: String) : Parcelable {
    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(id)
        dest.writeString(title)
    }

    private constructor(`in`: Parcel) :
            this(id = `in`.readString(),
                    title = `in`.readString())

    companion object {

        val CREATOR: Parcelable.Creator<ButtonDescription> = object : Parcelable.Creator<ButtonDescription> {
            override fun createFromParcel(`in`: Parcel): ButtonDescription {
                return ButtonDescription(`in`)
            }

            override fun newArray(size: Int): Array<ButtonDescription?> {
                return arrayOfNulls(size)
            }
        }
    }
}
